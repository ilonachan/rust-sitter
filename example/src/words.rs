#[rust_sitter::grammar("words")]
pub mod grammar {
    #[rust_sitter::language]
    #[derive(Debug)]
    pub struct Words {
        #[rust_sitter::leaf(text = r"if")]
        _keyword: (),
        #[rust_sitter::word]
        #[rust_sitter::leaf(pattern = r"[a-z_]+", transform = |v| v.to_string())]
        _word: String,
    }

    #[rust_sitter::extra]
    struct Whitespace {
        #[rust_sitter::leaf(pattern = r"\s")]
        _whitespace: (),
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn words_grammar() {
        insta::assert_debug_snapshot!(grammar::parse("if", None));
        insta::assert_debug_snapshot!(grammar::parse("hello", None));
        insta::assert_debug_snapshot!(grammar::parse("ifhello", None));
        insta::assert_debug_snapshot!(grammar::parse("if hello", None));
    }
}
