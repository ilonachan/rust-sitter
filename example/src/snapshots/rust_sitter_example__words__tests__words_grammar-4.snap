---
source: example/src/words.rs
expression: "grammar::parse(\"if hello\", None)"
---
Ok(
    (
        Words {
            _keyword: (),
            _word: "hello",
        },
        EditTree,
    ),
)
